src/components/autocomplete/js/autocompleteDirective.js

```diff
@@ -228,66 +230,54 @@ function MdAutocomplete () {
       }
 
       function getInputElement () {
+        var commonInputAttrs =
+          'type="search" \n' +
+          'name="{{inputName}}" \n' +
+          'autocomplete="off" \n' +
+          'ng-required="$mdAutocompleteCtrl.isRequired" \n' +
+          'ng-disabled="$mdAutocompleteCtrl.isDisabled" \n' +
+          'ng-model="$mdAutocompleteCtrl.scope.searchText" \n' +
+          'ng-readonly="$mdAutocompleteCtrl.isReadonly" \n' +
+          'ng-keydown="$mdAutocompleteCtrl.keydown($event)" \n' +
+          'ng-blur="$mdAutocompleteCtrl.blur()" \n' +
+          'ng-focus="$mdAutocompleteCtrl.focus()" \n' +
+          'ng-class="::inputClass" \n' +
+          'aria-autocomplete="list" \n' +
+          'aria-activedescendant="" \n' +
+          'aria-expanded="{{!$mdAutocompleteCtrl.hidden}}" \n' +
+          'aria-owns="ul-{{$mdAutocompleteCtrl.id}}" \n' +
+          (attr.mdSelectOnFocus != null ? 'md-select-on-focus="" \n' : '') +
+          (tabindex != null ? 'tabindex="' + tabindex + '" \n' : '');
+
         if (attr.mdFloatingLabel) {
-          return '\
-            <md-input-container flex ng-if="floatingLabel">\
-              <label>{{floatingLabel}}</label>\
-              <input type="search"\
-                  ' + (tabindex != null ? 'tabindex="' + tabindex + '"' : '') + '\
-                  id="{{ inputId || \'fl-input-\' + $mdAutocompleteCtrl.id }}"\
-                  name="{{inputName}}"\
-                  autocomplete="off"\
-                  ng-required="$mdAutocompleteCtrl.isRequired"\
-                  ng-readonly="$mdAutocompleteCtrl.isReadonly"\
-                  ng-minlength="inputMinlength"\
-                  ng-maxlength="inputMaxlength"\
-                  ng-disabled="$mdAutocompleteCtrl.isDisabled"\
-                  ng-model="$mdAutocompleteCtrl.scope.searchText"\
-                  ng-keydown="$mdAutocompleteCtrl.keydown($event)"\
-                  ng-blur="$mdAutocompleteCtrl.blur()"\
-                  ' + (attr.mdNoAsterisk != null ? 'md-no-asterisk="' + attr.mdNoAsterisk + '"' : '') + '\
-                  ng-focus="$mdAutocompleteCtrl.focus()"\
-                  aria-owns="ul-{{$mdAutocompleteCtrl.id}}"\
-                  ' + (attr.mdSelectOnFocus != null ? 'md-select-on-focus=""' : '') + '\
-                  aria-label="{{floatingLabel}}"\
-                  aria-autocomplete="list"\
-                  aria-haspopup="true"\
-                  aria-activedescendant=""\
-                  aria-expanded="{{!$mdAutocompleteCtrl.hidden}}"/>\
-              <div md-autocomplete-parent-scope md-autocomplete-replace>' + leftover + '</div>\
-            </md-input-container>';
+          return '' +
+            '<md-input-container flex ng-if="floatingLabel"> \n' +
```

Comment:
Nah I was referring to concatenating instead of escaping stuff. Otherwise you're right about the newlines, they just make it a little more convenient if uou need to log something.